src/blocking/i2c.rs

```diff
@@ -5,8 +5,32 @@
 //! Operations on 10-bit slave addresses are not supported by the API yet (but applications might
 //! be able to emulate some operations).
 
+use crate::private;
+
+/// Address mode (7-bit / 10-bit)
+///
+/// Note: This trait is sealed and should not be implemented outside of this crate.
+pub trait AddressMode: private::Sealed {
+    /// Address value type
+    type Address;
+}
+
+/// 7-bit address mode type
+pub struct SevenBitAddress(());
```

Comment:
Why the unit type inside the struct?

Reply:
It is a private data member. That way users cannot create an instance. It should only be a marker type.

Reply:
I'm aware what it does, just not sure about the purpose. Which bad things coul happen if someone decided to instantiate a `SevenBitAddress`?

Reply:
It is possible to implement a trait for `SevenBitAddress` but with a private data member it is at least not possible to create an instance and call stateful methods.
I do not know if this would be useful for somebody, though.